cranelift/codegen/src/isa/unwind/winarm64.rs

```diff
@@ -0,0 +1,308 @@
+//! Windows Arm64 ABI unwind information.
+
+use alloc::vec::Vec;
+#[cfg(feature = "enable-serde")]
+use serde_derive::{Deserialize, Serialize};
+
+use crate::binemit::CodeOffset;
+use crate::isa::unwind::UnwindInst;
+use crate::result::CodegenResult;
+
+use super::Writer;
+
+/// The supported unwind codes for the Arm64 Windows ABI.
+///
+/// See: <https://learn.microsoft.com/en-us/cpp/build/arm64-exception-handling>
+/// Only what is needed to describe the prologues generated by the Cranelift AArch64 ISA are represented here.
+#[allow(dead_code)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub(crate) enum UnwindCode {
+    /// Save int register, or register pair.
+    SaveReg {
+        reg: u8,
+        stack_offset: u16,
+        is_pair: bool,
+    },
+    /// Save floating point register, or register pair.
+    SaveFReg {
+        reg: u8,
+        stack_offset: u16,
+        is_pair: bool,
+    },
+    /// Save frame-pointer register (X29) and LR register pair.
+    SaveFpLrPair {
+        stack_offset: u16,
+    },
+    SmallStackAlloc {
+        size: u16,
+    },
+    MediumStackAlloc {
+        size: u16,
+    },
+    LargeStackAlloc {
+        size: u32,
+    },
+    /// PAC sign the LR register.
+    PacSignLr,
+    /// Set the frame-pointer register to the stack-pointer register.
+    SetFp,
+    /// Set the frame-pointer register to the stack-pointer register with an
+    /// offset.
+    AddFp {
+        offset: u16,
+    },
+}
```

Comment:
Can the names of each of these individual codes match the name in the link above? Specifically `SmallStackAlloc` is called `alloc_s` there so I'd expect it to be `AllocS` here.

Also, it would be nice to have doc comments for the variants that are missing them. For example, the difference between small vs medium stack alloc isn't immediately clear to the reader because they have the same `u16` data representation.

Reply:
Done: Renamed the alloc items and added a comment to explain the max size of each one.